src/plugins/intel_gpu/src/plugin/ops/roll.cpp

```diff
@@ -23,23 +23,23 @@ void CreateRollOp(ProgramBuilder& p, const std::shared_ptr<ov::op::v7::Roll>& op
 
     auto shift_constant = ov::as_type_ptr<ov::op::v0::Constant>(op->get_input_node_shared_ptr(1));
     OPENVINO_ASSERT(shift_constant != nullptr, "[GPU] Unsupported parameter nodes type in ", op_friendly_name, " (", op->get_type_name(), ")");
-    const auto shift_raw = shift_constant->cast_vector<int32_t>();
+    const auto shift_raw = shift_constant->cast_vector<cldnn::tensor::value_type>();
 
     auto axes_constant = ov::as_type_ptr<ov::op::v0::Constant>(op->get_input_node_shared_ptr(2));
     OPENVINO_ASSERT(axes_constant != nullptr, "[GPU] Unsupported parameter nodes type in ", op_friendly_name, " (", op->get_type_name(), ")");
-    auto axes_raw = axes_constant->cast_vector<int32_t>();
+    auto axes_raw = axes_constant->cast_vector<cldnn::tensor::value_type>();
 
     if (input_pshape.is_dynamic()) {
         const cldnn::roll roll_prim(layer_name, inputs.front(), shift_raw, axes_raw);
         p.add_primitive(*op, roll_prim);
     } else {
         const auto& input_shape = input_pshape.to_shape();
-        const auto rank = static_cast<int>(input_shape.size());
+        const auto rank = static_cast<cldnn::tensor::value_type>(input_shape.size());
```

Comment:
is it correct that `rank` is also `value_type`?

Reply:
Probably yes,
Below shift tensor uses rank variable for shift tensor value_type, and it used in cldnn::roll primitive constructor.
Use tensor::value_type in rank is safety than int32_t